src/earn/slice.ts

```diff
@@ -1,22 +1,26 @@
 import { PayloadAction, createSlice } from '@reduxjs/toolkit'
 import { REHYDRATE, RehydrateAction } from 'redux-persist'
-import { DepositInfo } from 'src/earn/types'
+import { DepositInfo, WithdrawInfo } from 'src/earn/types'
 import { getRehydratePayload } from 'src/redux/persist-helper'
 
+type Status = 'idle' | 'loading' | 'success' | 'error'
+
 interface State {
-  depositStatus: 'idle' | 'started' | 'success' | 'error'
+  depositStatus: Status
```

Comment:
Eventually have to handle multiple pools so deposit and withdraw status could need to be represented differently, but I think for the MVP this is fine.